lib/ffmprb/process.rb

```diff
@@ -58,7 +59,19 @@ def duck_audio(av_main_i, a_overlay_i, silence, av_main_o,
             roll in_main
 
             ducked_overlay_volume = {0.0 => volume_lo}
-            silence.each do |silent|
+
+            silent_fragments = silence.map &:clone
+            silent_fragments.each_cons(2) do |(silent, next_silent)|
+              silent_part = silent.overlaps || silent
+
+              if silent_part.end_at.nil? or (next_silent.start_at - silent_part.end_at) < audible_min
+                silent_part.end_at = next_silent.end_at
+                next_silent.overlaps = silent_part
+              end
+            end
+            silent_fragments.delete_if &:overlaps
+
+            silent_fragments.each do |silent|
               next  if silent.end_at && silent.start_at && (silent.end_at - silent.start_at) < silent_min
```

Comment:
Extract the silence preprocessing (both min. audible and min. silent filtering) into a method and write a unit test for it please. So the call above will be something like
`ducking_silence(silence).each do |silent|`
And then you're good to merge.